File: qt/gen_qbuttongroup.go
package qt

import (
	"unsafe"
)

type QButtonGroup struct {
	h          uintptr
	isSubclass bool
}

// NewQButtonGroup constructs a new QButtonGroup object.
func NewQButtonGroup() *QButtonGroup {
	g := newQButtonGroup(QButtonGroup_new())
	g.isSubclass = true
	return g
}

// NewQButtonGroup2 constructs a new QButtonGroup object.
func NewQButtonGroup2(parent *QObject) *QButtonGroup {
	g := newQButtonGroup(QButtonGroup_new2(parent.cPointer()))
	g.isSubclass = true
	return g
}

func (this *QButtonGroup) MetaObject() *QMetaObject {
	return newQMetaObject(QButtonGroup_MetaObject(this.h))
}

func (this *QButtonGroup) Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))
	return (unsafe.Pointer)(QButtonGroup_Metacast(this.h, param1_Cstring))
}

func QButtonGroup_Tr(s string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	var _ms struct_miqt_string = QButtonGroup_Tr(s_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QButtonGroup) SetExclusive(exclusive bool) {
	QButtonGroup_SetExclusive(this.h, (bool)(exclusive))
}

func (this *QButtonGroup) Exclusive() bool {
	return (bool)(QButtonGroup_Exclusive(this.h))
}

func (this *QButtonGroup) AddButton(param1 *QAbstractButton) {
	QButtonGroup_AddButton(this.h, param1.cPointer())
}

func (this *QButtonGroup) RemoveButton(param1 *QAbstractButton) {
	QButtonGroup_RemoveButton(this.h, param1.cPointer())
}

func (this *QButtonGroup) Buttons() []*QAbstractButton {
	var _ma struct_miqt_array = QButtonGroup_Buttons(this.h)
	_ret := make([]*QAbstractButton, int(_ma.len))
	_outCast := (*[0xffff]*QAbstractButton)(unsafe.Pointer(_ma.data)) // hey ya
	for i := 0; i < int(_ma.len); i++ {
		_ret[i] = newQAbstractButton(_outCast[i])
	}
	return _ret
}

func (this *QButtonGroup) CheckedButton() *QAbstractButton {
	return newQAbstractButton(QButtonGroup_CheckedButton(this.h))
}

func (this *QButtonGroup) Button(id int) *QAbstractButton {
	return newQAbstractButton(QButtonGroup_Button(this.h, (int)(id)))
}

func (this *QButtonGroup) SetId(button *QAbstractButton, id int) {
	QButtonGroup_SetId(this.h, button.cPointer(), (int)(id))
}

func (this *QButtonGroup) Id(button *QAbstractButton) int {
	return (int)(QButtonGroup_Id(this.h, button.cPointer()))
}

func (this *QButtonGroup) CheckedId() int {
	return (int)(QButtonGroup_CheckedId(this.h))
}

func (this *QButtonGroup) ButtonClicked(param1 *QAbstractButton) {
	QButtonGroup_ButtonClicked(this.h, param1.cPointer())
}

func (this *QButtonGroup) OnButtonClicked(slot func(param1 *QAbstractButton)) {
	QButtonGroup_connect_ButtonClicked(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_ButtonClicked
func miqt_exec_callback_QButtonGroup_ButtonClicked(cb intptr_t, param1 *QAbstractButton) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 *QAbstractButton))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQAbstractButton(param1)

	gofunc(slotval1)
}

func (this *QButtonGroup) ButtonPressed(param1 *QAbstractButton) {
	QButtonGroup_ButtonPressed(this.h, param1.cPointer())
}

func (this *QButtonGroup) OnButtonPressed(slot func(param1 *QAbstractButton)) {
	QButtonGroup_connect_ButtonPressed(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_ButtonPressed
func miqt_exec_callback_QButtonGroup_ButtonPressed(cb intptr_t, param1 *QAbstractButton) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 *QAbstractButton))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQAbstractButton(param1)

	gofunc(slotval1)
}

func (this *QButtonGroup) ButtonReleased(param1 *QAbstractButton) {
	QButtonGroup_ButtonReleased(this.h, param1.cPointer())
}

func (this *QButtonGroup) OnButtonReleased(slot func(param1 *QAbstractButton)) {
	QButtonGroup_connect_ButtonReleased(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_ButtonReleased
func miqt_exec_callback_QButtonGroup_ButtonReleased(cb intptr_t, param1 *QAbstractButton) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 *QAbstractButton))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQAbstractButton(param1)

	gofunc(slotval1)
}

func (this *QButtonGroup) ButtonToggled(param1 *QAbstractButton, param2 bool) {
	QButtonGroup_ButtonToggled(this.h, param1.cPointer(), (bool)(param2))
}

func (this *QButtonGroup) OnButtonToggled(slot func(param1 *QAbstractButton, param2 bool)) {
	QButtonGroup_connect_ButtonToggled(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_ButtonToggled
func miqt_exec_callback_QButtonGroup_ButtonToggled(cb intptr_t, param1 *QAbstractButton, param2 bool) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 *QAbstractButton, param2 bool))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := newQAbstractButton(param1)

	slotval2 := (bool)(param2)

	gofunc(slotval1, slotval2)
}

func (this *QButtonGroup) IdClicked(param1 int) {
	QButtonGroup_IdClicked(this.h, (int)(param1))
}

func (this *QButtonGroup) OnIdClicked(slot func(param1 int)) {
	QButtonGroup_connect_IdClicked(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_IdClicked
func miqt_exec_callback_QButtonGroup_IdClicked(cb intptr_t, param1 int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(param1)

	gofunc(slotval1)
}

func (this *QButtonGroup) IdPressed(param1 int) {
	QButtonGroup_IdPressed(this.h, (int)(param1))
}

func (this *QButtonGroup) OnIdPressed(slot func(param1 int)) {
	QButtonGroup_connect_IdPressed(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_IdPressed
func miqt_exec_callback_QButtonGroup_IdPressed(cb intptr_t, param1 int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(param1)

	gofunc(slotval1)
}

func (this *QButtonGroup) IdReleased(param1 int) {
	QButtonGroup_IdReleased(this.h, (int)(param1))
}

func (this *QButtonGroup) OnIdReleased(slot func(param1 int)) {
	QButtonGroup_connect_IdReleased(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_IdReleased
func miqt_exec_callback_QButtonGroup_IdReleased(cb intptr_t, param1 int) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 int))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(param1)

	gofunc(slotval1)
}

func (this *QButtonGroup) IdToggled(param1 int, param2 bool) {
	QButtonGroup_IdToggled(this.h, (int)(param1), (bool)(param2))
}

func (this *QButtonGroup) OnIdToggled(slot func(param1 int, param2 bool)) {
	QButtonGroup_connect_IdToggled(this.h, intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_IdToggled
func miqt_exec_callback_QButtonGroup_IdToggled(cb intptr_t, param1 int, param2 bool) {
	gofunc, ok := cgo.Handle(cb).Value().(func(param1 int, param2 bool))
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	// Convert all CABI parameters to Go parameters
	slotval1 := (int)(param1)

	slotval2 := (bool)(param2)

	gofunc(slotval1, slotval2)
}

func QButtonGroup_Tr2(s string, c string) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QButtonGroup_Tr2(s_Cstring, c_Cstring)
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func QButtonGroup_Tr3(s string, c string, n int) string {
	s_Cstring := CString(s)
	defer free(unsafe.Pointer(s_Cstring))
	c_Cstring := CString(c)
	defer free(unsafe.Pointer(c_Cstring))
	var _ms struct_miqt_string = QButtonGroup_Tr3(s_Cstring, c_Cstring, (int)(n))
	_ret := GoStringN(_ms.data, int(int64(_ms.len)))
	free(unsafe.Pointer(_ms.data))
	return _ret
}

func (this *QButtonGroup) AddButton2(param1 *QAbstractButton, id int) {
	QButtonGroup_AddButton2(this.h, param1.cPointer(), (int)(id))
}

func (this *QButtonGroup) callVirtualBase_MetaObject() *QMetaObject {
	return newQMetaObject(QButtonGroup_virtualbase_MetaObject(unsafe.Pointer(this.h)))
}

func (this *QButtonGroup) OnMetaObject(slot func(super func() *QMetaObject) *QMetaObject) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QButtonGroup_override_virtual_MetaObject(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_MetaObject
func miqt_exec_callback_QButtonGroup_MetaObject(self QButtonGroup, cb intptr_t) *QMetaObject {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func() *QMetaObject) *QMetaObject)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}

	virtualReturn := gofunc((&QButtonGroup{h: self}).callVirtualBase_MetaObject)

	return virtualReturn.cPointer()
}

func (this *QButtonGroup) callVirtualBase_Metacast(param1 string) unsafe.Pointer {
	param1_Cstring := CString(param1)
	defer free(unsafe.Pointer(param1_Cstring))

	return (unsafe.Pointer)(QButtonGroup_virtualbase_Metacast(unsafe.Pointer(this.h), param1_Cstring))
}

func (this *QButtonGroup) OnMetacast(slot func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer) {
	if !this.isSubclass {
		panic("miqt: can only override virtual methods for directly constructed types")
	}
	QButtonGroup_override_virtual_Metacast(unsafe.Pointer(this.h), intptr_t(cgo.NewHandle(slot)))
}

//export miqt_exec_callback_QButtonGroup_Metacast
func miqt_exec_callback_QButtonGroup_Metacast(self QButtonGroup, cb intptr_t, param1 *const_char) unsafe.Pointer {
	gofunc, ok := cgo.Handle(cb).Value().(func(super func(param1 string) unsafe.Pointer, param1 string) unsafe.Pointer)
	if !ok {
		panic("miqt: callback of non-callback type (heap corruption?)")
	}
	// Convert all CABI parameters to Go parameters
	param1_ret := param1
	slotval1 := GoString(param1_ret)
	virtualReturn := gofunc((&QButtonGroup{h: self}).callVirtualBase_Metacast, slotval1)

	return virtualReturn
}

File: qt/gen_qbuttongroup.cpp
// +build ignore

#include <QAbstractButton>
#include <QButtonGroup>
#include <QList>
#include <QMetaObject>
#include <QObject>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <qbuttongroup.h>
#include "gen_qbuttongroup.h"
class MiqtVirtualQButtonGroup : public virtual QButtonGroup {
public:
MiqtVirtualQButtonGroup(): QButtonGroup() {};
MiqtVirtualQButtonGroup(QObject* parent): QButtonGroup(parent) {};

virtual ~MiqtVirtualQButtonGroup() = default;
// cgo.Handle value for overwritten implementation
	intptr_t handle__MetaObject = 0;
// Subclass to allow providing a Go implementation
	virtual const QMetaObject* metaObject() const override {
if (handle__MetaObject == 0) {
return QButtonGroup::metaObject();
}
QMetaObject* callback_return_value = miqt_exec_callback_QButtonGroup_MetaObject(const_cast<MiqtVirtualQButtonGroup*>(this), handle__MetaObject);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	QMetaObject* virtualbase_MetaObject() const {

		return (QMetaObject*) QButtonGroup::metaObject();

	}
// cgo.Handle value for overwritten implementation
	intptr_t handle__Metacast = 0;
// Subclass to allow providing a Go implementation
	virtual void* qt_metacast(const char* param1) override {
if (handle__Metacast == 0) {
return QButtonGroup::qt_metacast(param1);
}
const char* sigval1 = (const char*) param1;

		void* callback_return_value = miqt_exec_callback_QButtonGroup_Metacast(this, handle__Metacast, sigval1);

		return callback_return_value;
	}
// Wrapper to allow calling protected method
	void* virtualbase_Metacast(const char* param1) {

		return QButtonGroup::qt_metacast(param1);

	}
};
QButtonGroup* QButtonGroup_new() {
return new MiqtVirtualQButtonGroup();
}
QButtonGroup* QButtonGroup_new2(QObject* parent) {
return new MiqtVirtualQButtonGroup(parent);
}
void QButtonGroup_virtbase(QButtonGroup* src
, QObject** outptr_QObject
) {
*outptr_QObject = static_cast<QObject*>(src);
}
QMetaObject* QButtonGroup_MetaObject(const QButtonGroup* self) {
	return (QMetaObject*) self->metaObject();
}
void* QButtonGroup_Metacast(QButtonGroup* self, const char* param1) {
	return self->qt_metacast(param1);
}
struct miqt_string QButtonGroup_Tr(const char* s) {
	QString _ret = QButtonGroup::tr(s);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QButtonGroup_SetExclusive(QButtonGroup* self, bool exclusive) {
	self->setExclusive(exclusive);
}
bool QButtonGroup_Exclusive(const QButtonGroup* self) {
	return self->exclusive();
}
void QButtonGroup_AddButton(QButtonGroup* self, QAbstractButton* param1) {
	self->addButton(param1);
}
void QButtonGroup_RemoveButton(QButtonGroup* self, QAbstractButton* param1) {
	self->removeButton(param1);
}
struct miqt_array /* of QAbstractButton* */  QButtonGroup_Buttons(const QButtonGroup* self) {
	QList<QAbstractButton *> _ret = self->buttons();
	// Convert QList<> from C++ memory to manually-managed C memory
	QAbstractButton** _arr = static_cast<QAbstractButton**>(malloc(sizeof(QAbstractButton*) * _ret.length()));
	for (size_t i = 0, e = _ret.length(); i < e; ++i) {
		_arr[i] = _ret[i];
	}
	struct miqt_array _out;
	_out.len = _ret.length();
	_out.data = static_cast<void*>(_arr);
	return _out;
}
QAbstractButton* QButtonGroup_CheckedButton(const QButtonGroup* self) {
	return self->checkedButton();
}
QAbstractButton* QButtonGroup_Button(const QButtonGroup* self, int id) {
	return self->button(static_cast<int>(id));
}
void QButtonGroup_SetId(QButtonGroup* self, QAbstractButton* button, int id) {
	self->setId(button, static_cast<int>(id));
}
int QButtonGroup_Id(const QButtonGroup* self, QAbstractButton* button) {
	return self->id(button);
}
int QButtonGroup_CheckedId(const QButtonGroup* self) {
	return self->checkedId();
}
void QButtonGroup_ButtonClicked(QButtonGroup* self, QAbstractButton* param1) {
	self->buttonClicked(param1);
}
void QButtonGroup_connect_ButtonClicked(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(QAbstractButton*)>(&QButtonGroup::buttonClicked), self, [=](QAbstractButton* param1) {
		QAbstractButton* sigval1 = param1;
		miqt_exec_callback_QButtonGroup_ButtonClicked(slot, sigval1);
	});
}
void QButtonGroup_ButtonPressed(QButtonGroup* self, QAbstractButton* param1) {
	self->buttonPressed(param1);
}
void QButtonGroup_connect_ButtonPressed(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(QAbstractButton*)>(&QButtonGroup::buttonPressed), self, [=](QAbstractButton* param1) {
		QAbstractButton* sigval1 = param1;
		miqt_exec_callback_QButtonGroup_ButtonPressed(slot, sigval1);
	});
}
void QButtonGroup_ButtonReleased(QButtonGroup* self, QAbstractButton* param1) {
	self->buttonReleased(param1);
}
void QButtonGroup_connect_ButtonReleased(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(QAbstractButton*)>(&QButtonGroup::buttonReleased), self, [=](QAbstractButton* param1) {
		QAbstractButton* sigval1 = param1;
		miqt_exec_callback_QButtonGroup_ButtonReleased(slot, sigval1);
	});
}
void QButtonGroup_ButtonToggled(QButtonGroup* self, QAbstractButton* param1, bool param2) {
	self->buttonToggled(param1, param2);
}
void QButtonGroup_connect_ButtonToggled(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(QAbstractButton*, bool)>(&QButtonGroup::buttonToggled), self, [=](QAbstractButton* param1, bool param2) {
		QAbstractButton* sigval1 = param1;
		bool sigval2 = param2;
		miqt_exec_callback_QButtonGroup_ButtonToggled(slot, sigval1, sigval2);
	});
}
void QButtonGroup_IdClicked(QButtonGroup* self, int param1) {
	self->idClicked(static_cast<int>(param1));
}
void QButtonGroup_connect_IdClicked(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(int)>(&QButtonGroup::idClicked), self, [=](int param1) {
		int sigval1 = param1;
		miqt_exec_callback_QButtonGroup_IdClicked(slot, sigval1);
	});
}
void QButtonGroup_IdPressed(QButtonGroup* self, int param1) {
	self->idPressed(static_cast<int>(param1));
}
void QButtonGroup_connect_IdPressed(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(int)>(&QButtonGroup::idPressed), self, [=](int param1) {
		int sigval1 = param1;
		miqt_exec_callback_QButtonGroup_IdPressed(slot, sigval1);
	});
}
void QButtonGroup_IdReleased(QButtonGroup* self, int param1) {
	self->idReleased(static_cast<int>(param1));
}
void QButtonGroup_connect_IdReleased(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(int)>(&QButtonGroup::idReleased), self, [=](int param1) {
		int sigval1 = param1;
		miqt_exec_callback_QButtonGroup_IdReleased(slot, sigval1);
	});
}
void QButtonGroup_IdToggled(QButtonGroup* self, int param1, bool param2) {
	self->idToggled(static_cast<int>(param1), param2);
}
void QButtonGroup_connect_IdToggled(QButtonGroup* self, intptr_t slot) {
	MiqtVirtualQButtonGroup::connect(self, static_cast<void (QButtonGroup::*)(int, bool)>(&QButtonGroup::idToggled), self, [=](int param1, bool param2) {
		int sigval1 = param1;
		bool sigval2 = param2;
		miqt_exec_callback_QButtonGroup_IdToggled(slot, sigval1, sigval2);
	});
}
struct miqt_string QButtonGroup_Tr2(const char* s, const char* c) {
	QString _ret = QButtonGroup::tr(s, c);
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
struct miqt_string QButtonGroup_Tr3(const char* s, const char* c, int n) {
	QString _ret = QButtonGroup::tr(s, c, static_cast<int>(n));
	// Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
	QByteArray _b = _ret.toUtf8();
	struct miqt_string _ms;
	_ms.len = _b.length();
	_ms.data = static_cast<char*>(malloc(_ms.len));
	memcpy(_ms.data, _b.data(), _ms.len);
	return _ms;
}
void QButtonGroup_AddButton2(QButtonGroup* self, QAbstractButton* param1, int id) {
	self->addButton(param1, static_cast<int>(id));
}
void QButtonGroup_override_virtual_MetaObject(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQButtonGroup*>( (QButtonGroup*)(self) )->handle__MetaObject = slot;
}
QMetaObject* QButtonGroup_virtualbase_MetaObject(const void* self) {
	return ( (const MiqtVirtualQButtonGroup*)(self) )->virtualbase_MetaObject();
}
void QButtonGroup_override_virtual_Metacast(void* self, intptr_t slot) {
	dynamic_cast<MiqtVirtualQButtonGroup*>( (QButtonGroup*)(self) )->handle__Metacast = slot;
}
void* QButtonGroup_virtualbase_Metacast(void* self, const char* param1) {
	return ( (MiqtVirtualQButtonGroup*)(self) )->virtualbase_Metacast(param1);
}
void QButtonGroup_Delete(QButtonGroup* self, bool isSubclass) {
	if (isSubclass) {
		delete dynamic_cast<MiqtVirtualQButtonGroup*>( self );
	} else {
		delete self;
	}
}
